native/src/modules/endpoint/sagas/loadCityContent.js

```diff
@@ -54,74 +54,88 @@ export default function * loadCityContent (
-    call(loadPois, newCity, newLanguage, featureFlags.pois, dataContainer, shouldUpdate)
-  ])
-
-  const netInfo = yield call(NetInfo.fetch)
-  const isCellularConnection = netInfo.type === 'cellular'
-
-  // fetchResourceCache should be callable independent of content updates. Even if loadCategories, loadEvents,
-  // loadLanguages did not update the dataContainer this is needed. In case the previous call to fetchResourceCache
-  // failed to download some resources an other call could fix this and download missing files.
-  if (criterion.shouldRefreshResources() && !isCellularConnection) {
-    const resourceURLFinder = new ResourceURLFinder(buildConfig().allowedHostNames)
-    resourceURLFinder.init()
-
-    const fetchMap = resourceURLFinder.buildFetchMap(
-      categoriesMap.toArray().concat(events),
-      (url, urlHash) => buildResourceFilePath(url, newCity, urlHash)
-    )
-
-    resourceURLFinder.finalize()
-    yield call(fetchResourceCache, newCity, newLanguage, fetchMap, dataContainer)
-  }
+    const { featureFlags } = buildConfig()
+    const [categoriesMap, events] = yield all([
+      call(loadCategories, newCity, newLanguage, dataContainer, shouldUpdate),
+      call(loadEvents, newCity, newLanguage, cityModel.eventsEnabled, dataContainer, shouldUpdate),
+      call(loadPois, newCity, newLanguage, featureFlags.pois, dataContainer, shouldUpdate)
+    ])
+
+    const netInfo = yield call(NetInfo.fetch)
+    const isCellularConnection = netInfo.type === 'cellular'
+
+    // fetchResourceCache should be callable independent of content updates. Even if loadCategories, loadEvents,
+    // loadLanguages did not update the dataContainer this is needed. In case the previous call to fetchResourceCache
+    // failed to download some resources an other call could fix this and download missing files.
+    if (criterion.shouldRefreshResources() && !isCellularConnection) {
+      const resourceURLFinder = new ResourceURLFinder(buildConfig().allowedHostNames)
+      resourceURLFinder.init()
+
+      const fetchMap = resourceURLFinder.buildFetchMap(
+        categoriesMap.toArray().concat(events),
+        (url, urlHash) => buildResourceFilePath(url, newCity, urlHash)
+      )
+
+      resourceURLFinder.finalize()
+      yield call(fetchResourceCache, newCity, newLanguage, fetchMap, dataContainer)
+    }
 
-  if (shouldUpdate) {
-    yield call(dataContainer.setLastUpdate, newCity, newLanguage, moment())
-  }
+    if (!shouldUpdate && lastUpdate) {
+      yield call(dataContainer.setLastUpdate, newCity, newLanguage, lastUpdate)
+    } else {
+      yield call(dataContainer.setLastUpdate, newCity, newLanguage, moment())
+    }
 
-  return true
+    return true
+  } catch (e) {
+    // If any error occurs, we have to store the old value for lastUpdate again
+    if (lastUpdate) {
+      yield call(dataContainer.setLastUpdate, newCity, newLanguage, lastUpdate)
+    }
+    throw e
+  }
```

Comment:
This should be the same as https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch:
```suggestion
  } finally (e) {
    // If any error occurs, we have to store the old value for lastUpdate again
    if (lastUpdate) {
      yield call(dataContainer.setLastUpdate, newCity, newLanguage, lastUpdate)
    }
  }
```

Reply:
Forgot this existed :smile: